src/Grammar.js

```diff
@@ -169,7 +164,7 @@ Grammar.prototype = {
       var body = self.ruleBodies[ruleName];
 
       var operation;
-      if (self.superGrammar.ruleBodies[ruleName]) {
+      if (self.superGrammar && self.superGrammar.ruleBodies[ruleName]) {
```

Comment:
Would you mind adding a `isExtendingOrOverriding` variable that holds the value of this condition, and then...